quinn/keyword_finder.py

```diff
@@ -41,41 +41,53 @@
 ]
 
 
-def search_file(path: str, keywords: list[str] = default_keywords) -> None:
+def search_file(path: str, keywords: list[str] = default_keywords) -> dict[str, dict[str, int]]:
```

Comment:
I would suggest to create a simple `dataclass` to handle such a complex output. Like:
```python
@dataclass
class SearchResult:
  file_path: str
  word_count: dict[str, int]
```

Reply:
@SemyonSinchenko yeah I agree that named values would be better than making the path the dictionary key. 

I tried out a `TypedDict` and a `dataclass` implementation - do you have a strong preference for  `dataclasses`? The `TypedDict` behaves more like a dictionary (ex: results['word_count']['rdd'] works, but that has to be changed to results.word_count['rdd'] in a `dataclass`.